### PR TITLE
Remove modelAsExtensible, add oldModelAsString

### PR DESCRIPTION
**Pull request by dsgouda**
Replacing the `modelAsExtensible` switch with `oldModelAsString`
This means we generate the extensible enum implementation by default for 
`"modelAsString": true` and generate the legacy non extensible string implementation for enums only if we see
`"modelAsString": true` and `"oldModelAsString":true`

Will open PRs for `autorest.modeler` and `autorest.csharp` once `autorest.common` and `autorest.testserver` changes are merged

### CODE REVIEW

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by lmazuel** [comment]
@olydis @dsgouda Does this mean that by default, no generator should fail on unkwon value from the RestAPI?

**Comment by fearthecowboy** [comment]
@lmazuel  

This shouldn't have any affect on non-c# generators; we're just making it so that c#'s extensible enum is used when `modelAsString` is true (instead of just simple strings) and still allow a back door to using the simple strings if absolutely required (adding oldModelAsString)

This shouldn't change anything for you; `modelAsString : true ` should tell you that you should accept/allow unknown values, `modelAsString : false`  indicates that you can be strict. If that's different than what you do now, we should pause for a moment and see why...

**Comment by lmazuel** [comment]
@fearthecowboy gotcha thanks!

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by fearthecowboy** [comment]
> linux-x64 restart
> win32-x64 restart

**Comment by fearthecowboy** [comment]
> linux-x64 restart

**Comment by fearthecowboy** [comment]
> win32-x64 restart

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

File: Samples/test/deprecated/shell/stderr.txt [hunk 1]
```diff
@@ -1 +1,11 @@
-
+    - /Samples/test/deprecated/deprecated.yaml:226:4 ($.definitions.Enum3No["x-ms-enum"])
+    - /Samples/test/deprecated/deprecated.yaml:235:4 ($.definitions.Enum3Yes["x-ms-enum"])
+    - /Samples/test/deprecated/deprecated.yaml:245:4 ($.definitions.Enum3Custom["x-ms-enum"])
+    - /Samples/test/deprecated/deprecated.yaml:255:4 ($.definitions.Enum3Replaced["x-ms-enum"])
+ERROR: Schema violation: Additional properties not allowed: modelAsExtensible
```
Comment (fearthecowboy) [inline Samples/test/deprecated/shell/stderr.txt]: Shouldn't we have taken `modelAsExtensible` out of the  model that causes this?

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Pull request merged**